src/types/modules/module.cpp

```diff
@@ -36,6 +36,30 @@ std::optional<ExpressionType> stringToExpressionType (const std::string& str)
 	return std::nullopt;
 }
 
+/**
+ * Finds an attribute with specified name for a structure or iterable base.
+ */
+std::optional<std::shared_ptr<Symbol>> _getExistingAttribute(Symbol* base, std::string& name)
+{
+	if (base)
+	{
+		switch (base->getType())
+		{
+			case Symbol::Type::Structure:
+				return ((StructureSymbol*)base)->getAttribute(name);
+			case Symbol::Type::Array:
+			case Symbol::Type::Dictionary:
+			{
+				auto structElement = ((IterableSymbol*)base)->getStructuredElementType();
```

Comment:
Casting base to derived class is always risky even when you check that you deal with the particular derived class instance. The best is to avoid such needs with different design when possible and when in real need it is better to use dynamic cast here. Is dynamic cast appliable here? Can we use something safer than (<>*) cast?

Reply:
We need access to the derived class because we want to be able to get and set the attribute of an iterable or structure base. I've tried to think of a way to avoid conversion, but that would lead to a duplication of *_addStruct()*- there would be one for a structure base and the other for an iterable base.
That would make it difficult to maintain or extend the code, should we attempt similar in-depth changes to the modules like I did in this PR, since functions in this module are usually called in a complicated circular dependency manner and further splitting of functionality would only make it worse.
Thank you for pointing out the possibility of using a dynamic_cast, I've adjusted the code to reflect that.